### DISCUSSION TITLE
lowest val/loss ckpt != highest val/Accuracy

### DISCUSSION THREAD

**Discussion by rahulvigneswaran** [Lightning-AI/pytorch-lightning#11896] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Am using the following callback,
```python
 checkpoint_callback = ModelCheckpoint(monitor='val/loss',
                                        mode='min',
                                        save_last=True,
                                        filename=cfg.CALLBACKS.FILENAME,
                                        auto_insert_metric_name=cfg.CALLBACKS.AUTO_INSERT_METRIC_NAME,
                                        dirpath=LOGGER_DIR)
```

![image](https://user-images.githubusercontent.com/37276766/153702380-cb8af624-4df0-4165-bcda-f27385bbb0aa.png)

I am not sure what is going wrong. Am using `F.cross_entropy` for loss.

**Comment by rahulvigneswaran** [comment, marked as answer]
- Lowest loss (say CE) will not always give you the highest accuracy. (https://kharshit.github.io/blog/2018/12/07/loss-vs-accuracy)
- So, always use the accuracy (or desired metric at hand) to select the best model instead of loss.

(This is probably common knowledge but this was news for me. Putting it out there so someone else doesn't bang their head for two days!)
